src/main/java/org/myrobotlab/framework/Service.java

```diff
@@ -2034,6 +2037,80 @@ synchronized public void startService() {
     }
   }
 
+  public void startPeers() {
+    log.info("starting peers");
+    Map<String, ServiceReservation> peers = null;
+
+    try {
+      Method method = this.getClass().getMethod("getMetaData");
+      ServiceType st = (ServiceType) method.invoke(null);
+      peers = st.getPeers();
+    } catch (Exception e) {
+    }
+
+    if (peers == null) {
+      return;
+    }
+    

```

Comment:
Why aren't we just calling Runtime.start here?  I'm not fully grokin' the need for the logic below.

Reply:
this is starting peers - there's a possibility of  not having getMetaData, nor having any peers ..  if this service has no peers, we are done ... 

I'd say that's better than an NPE